### PR TITLE
`getNamedExpressionsFromFormula` method

### LINKED ISSUE #1365: Feature: get a list of named expressions used by a formula
Create a method that returns a list of named expressions in a formula.

The method should take a cell address as parameter or the formula as string, whichever is better, easier or more consistent with the rest of the API.

Originally requested in https://github.com/handsontable/hyperformula/discussions/1077#discussioncomment-3879351

### PR DESCRIPTION
**Pull request by sequba**
### Context

- Added method `getNamedExpressionsFromFormula`

### How did you test your changes?

- [x] unit tests pass
- [x] new unit tests
- [x] check generated API ref

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [x] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [ ] Change to the documentation

### Related issues:
Fixes #1365 
https://github.com/handsontable/hyperformula/discussions/1077

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [x] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: src/HyperFormula.ts [hunk 1]
```diff
@@ -4104,6 +4104,42 @@ export class HyperFormula implements TypedEmitter {
...
+   * Return a list of named expressions used by a formula.
+   *
+   * @param {string} formulaString - A formula in a proper format, starting with `=`.
+   *
+   * @throws [[ExpectedValueOfTypeError]] if any of its basic type arguments is of wrong type.
+   * @throws [[NotAFormulaError]] when the provided string is not a valid formula (i.e., doesn't start with `=`).
+   *
+   * @example
+   * ```js
+   * // TODO
```
Comment (sequba) [inline src/HyperFormula.ts:4117]: TODO: add example for jsdoc

Comment (sequba) [reply src/HyperFormula.ts]: done

File: src/HyperFormula.ts [hunk 2]
```diff
@@ -4104,6 +4104,42 @@ export class HyperFormula implements TypedEmitter {
...
+    validateArgToType(formulaString, 'string', 'formulaString')
+    const { ast, dependencies } = this.extractTemporaryFormula(formulaString)
+
+    if (ast === undefined) {
+      throw new NotAFormulaError()
+    }
+
+    const namedExpressionDependencies = dependencies
+      .filter(dep => dep instanceof NamedExpressionDependency)
+      .map(namedExpr => (namedExpr as NamedExpressionDependency).name)
```
Comment (evanSe) [inline src/HyperFormula.ts:4136]: is it possible to avoid doing a `as NamedExpressionDependency` here? perhaps we can do some narrowing etc

**Review by evanSe — APPROVED** [review]

File: src/HyperFormula.ts (hunk 2 above)
Comment (sequba) [reply src/HyperFormula.ts]: Good point. I used type narrowing in the filter method

**Pull request merged**